kotest-assertions/kotest-assertions-core/src/commonMain/kotlin/io/kotest/matchers/string/matchers.kt

```diff
@@ -40,10 +40,11 @@ fun String?.shouldNotContainADigit(): String? {
 }
 
 fun containADigit() = neverNullMatcher<String> { value ->
+   val firstDigit = value.asSequence().withIndex().firstOrNull { it.value in '0'..'9' }
```

Comment:
I think you can maybe go `indexOfFirst { }` and get the index only?

If you want to use it on the description we could maybe destructure them instead
```suggestion
   val (firstDigitIndex, firstDigit) = value.asSequence().withIndex().firstOrNull { it.value in '0'..'9' }
```

Reply:
I also think this could benefit from being a "FirstDigitIndex" to being a "FirstNumberIndex"

Let's say

The string "I paid 200 already" != "I paid already"


The error message could be "Contained 200 at index 7". I don't know how important this would be, and we would also need to guard against having incredibly big numbers printed to the console, but if you haven't thought of this option yet, here it is :D

Reply:
> I also think this could benefit from being a "FirstDigitIndex" to being a "FirstNumberIndex"
> 
> Let's say
> 
> The string "I paid 200 already" != "I paid already"
> 
> The error message could be "Contained 200 at index 7", which would have more meaning than "Contained 2 at index 7". I don't know how important this would be, and we would also need to guard against having incredibly big numbers printed to the console, but if you haven't thought of this option yet, here it is :D

makes sense. I'll improve tomorrow, TY for calling it out!

Reply:
> I think you can maybe go `indexOfFirst { }` and get the index only?

done, and the code is cleaner. TY